bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml

```diff
@@ -110,64 +105,67 @@
   <build>
     <plugins>
       <plugin>
-        <artifactId>maven-assembly-plugin</artifactId>
-        <configuration>
-          <archive>
-            <manifest>
-              <mainClass>
-                ${mainclass}
-              </mainClass>
-            </manifest>
-          </archive>
-          <descriptorRefs>
-            <descriptorRef>jar-with-dependencies</descriptorRef>
-          </descriptorRefs>
-        </configuration>
+        <groupId>com.google.cloud.bigtable.test</groupId>
+        <artifactId>bigtable-build-helper</artifactId>
+        <version>2.7.3-SNAPSHOT</version> <!-- {x-version-update:bigtable-client-parent:current} -->
+
         <executions>
           <execution>
+            <id>verify-provided</id>
             <goals>
-              <goal>single</goal>
+              <goal>verify-provided-deps</goal>
             </goals>
-            <phase>package</phase>
           </execution>
         </executions>
       </plugin>
+
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-jar-plugin</artifactId>
-        <configuration>
-          <archive>
-            <manifest>
-              <mainClass>
-                ${mainclass}
-              </mainClass>
-            </manifest>
-          </archive>
-        </configuration>
-      </plugin>
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-enforcer-plugin</artifactId>
+        <artifactId>maven-shade-plugin</artifactId>
         <executions>
           <execution>
-            <id>enforce-banned-deps</id>
+            <id>shade</id>
             <goals>
-              <goal>enforce</goal>
+              <goal>shade</goal>
             </goals>
             <configuration>
-              <rules>
-                <bannedDependencies>
+              <createDependencyReducedPom>true</createDependencyReducedPom>
+              <filters>
+                <filter>
+                  <artifact>*:*</artifact>
```

Comment:
for my understanding, what's this doing?

Reply:
Its telling maven shade plugin that the exclude below apply to all deps